utils/zerovec/src/zerovec/mod.rs

```diff
@@ -80,55 +83,46 @@ use core::ops::Deref;
 /// Benchmarks can be found in the project repository, with some results found in the [crate-level documentation](crate).
 ///
 /// See [the design doc](https://github.com/unicode-org/icu4x/blob/main/utils/zerovec/design_doc.md) for more details.
-#[derive(Clone)]
-#[non_exhaustive]
-pub enum ZeroVec<'a, T>
+pub struct ZeroVec<'a, T>
 where
-    T: AsULE + ?Sized,
+    T: AsULE,
 {
-    /// An owned `ZeroVec<T>`. This will typically be constructed by [`ZeroVec::alloc_from_slice()`]
-    /// or by calling [`ZeroVec::to_mut()`]/[`ZeroVec::for_each_mut()`]/etc on [`ZeroVec::Borrowed`].
-    Owned(Vec<T::ULE>),
-
-    /// A borrowed `ZeroVec<T>`. This will typically be constructed by [`ZeroVec::parse_byte_slice()`],
-    /// [`ZeroVec::from_slice_or_alloc()`], or deserializers capable of doing zero-copy deserialization.
-    ///
-    /// If you already have a slice of `[T::ULE]`s, you can directly construct one of these.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use zerovec::ule::*;
-    /// use zerovec::ZeroVec;
-    ///
-    /// // The little-endian bytes correspond to the numbers on the following line.
-    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
-    /// let nums: &[RawBytesULE<2>] = &[
-    ///     211_u16.to_unaligned(),
-    ///     281_u16.to_unaligned(),
-    ///     421_u16.to_unaligned(),
-    ///     461_u16.to_unaligned(),
-    /// ];
-    ///
-    /// let zerovec = ZeroVec::<u16>::Borrowed(nums);
-    ///
-    /// assert!(matches!(zerovec, ZeroVec::Borrowed(_)));
-    /// assert_eq!(bytes, zerovec.as_bytes());
-    /// ```
-    Borrowed(&'a [T::ULE]),
+    /// Pointer to data
+    buf: *mut [T::ULE],
```

Comment:
Issue: Don't you need a manual `Drop` impl to clean this up?

Reply:
oh shoot, yeah